substrate/client/rpc-spec-v2/src/transaction/transaction_broadcast.rs

```diff
@@ -111,12 +113,21 @@ where
 				|notification| async move { notification.is_new_best.then_some(notification.hash) },
 			));
 
+		let broadcast_ids = self.broadcast_ids.clone();
+		let current_id = id.clone();
 		let broadcast_transaction_fut = async move {
 			// There is nothing we could do with an extrinsic of invalid format.
 			let Ok(decoded_extrinsic) = TransactionFor::<Pool>::decode(&mut &bytes[..]) else {
 				return;
 			};
 
+			// Save the tx hash to remove it later.
+			let tx_hash = pool.hash_of(&decoded_extrinsic);
+			broadcast_ids
+				.write()
+				.get_mut(&current_id)
+				.map(|state| state.tx_hash = Some(tx_hash));
```

Comment:
You can move this out of the future and store the `tx_hash` directly. Then you also don't need to use an `Option` for the the `tx_hash`.

Reply:
Yep, that makes sense! Thanks for the review!